vendor/bat-native-ads/src/bat/ads/internal/security/conversions/verifiable_conversion_envelope_info.cc

```diff
@@ -0,0 +1,39 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/security/conversions/verifiable_conversion_envelope_info.h"
+
+namespace ads {
+namespace security {
+
+VerifiableConversionEnvelopeInfo::VerifiableConversionEnvelopeInfo() = default;
+
+VerifiableConversionEnvelopeInfo::VerifiableConversionEnvelopeInfo(
+    const VerifiableConversionEnvelopeInfo& info) = default;
+
+VerifiableConversionEnvelopeInfo::~VerifiableConversionEnvelopeInfo() = default;
+
+bool VerifiableConversionEnvelopeInfo::operator==(
+    const VerifiableConversionEnvelopeInfo& rhs) const {
+  return algorithm == rhs.algorithm && ciphertext == rhs.ciphertext &&
+         ephemeral_public_key == rhs.ephemeral_public_key && nonce == rhs.nonce;
+}
+
+bool VerifiableConversionEnvelopeInfo::operator!=(
+    const VerifiableConversionEnvelopeInfo& rhs) const {
+  return !(*this == rhs);
+}
+
+bool VerifiableConversionEnvelopeInfo::IsValid() const {
+  if (algorithm.empty() || ciphertext.empty() || ephemeral_public_key.empty() ||
```

Comment:
Is this read by the recipient as well? If so it should validate the following:

1. the algorithm isnt just empty but it matches: `kAlgorithm`. We should reject any message that doesn't contain that property exactly.
2. the message is the correct length (they should all be the same after encoding yes?)

Nonce and EPK lengths should be validated by the C lib.

Reply:
this is just "dumb" logic of the info object - it shouldn't contain business logic and is just how we use info objects in the ads lib. The checks are all in `conversions_util.cc`. Let me know if that works @orspetol

Reply:
and to answer your question, this is never read by the recipient. It is just used for envelope creation and unittests.